src/nat/tool/nvidia_rag.py

```
@@ -58,31 +122,24 @@ async def nvidia_rag_tool(config: NVIDIARAGToolConfig, builder: Builder):
-                            "document_full_text": ret["content"],
-                            "score_rerank": ret["score"]
-                        },
-                        type="Document",
-                    ) for ret in output["chunks"]
-                ]
-
-                parsed_output = config.document_separator.join(
+                try:
+                    docs: list[Document] = parse_rag_response(response.json())
+                except (ValidationError, ValueError) as e:
+                    logger.error("RAG response validation failed: %s", e)
```

Ruff (0.14.11)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)